### PR TITLE
ci: add node and install ci-related libraries

### PR DESCRIPTION
**Pull request by 0xernesto**

### CODE REVIEW

File: script/Create2Factory.s.sol [hunk 1]
```diff
@@ -1,10 +1,10 @@
 // SPDX-License-Identifier: UNLICENSED
```
Comment (cucupac) [inline script/Create2Factory.s.sol:1]: Change the name of this script file.

**Review by cucupac — CHANGES_REQUESTED** [review]
See comments.

File: src/PredictiveDeployer.sol [hunk 2]
```diff
@@ -91,7 +95,7 @@ contract PredictiveDeployer is Initializable, UUPSUpgradeable, Ownable {
         address recoveredSigner = ECDSA.recover(expectedMessageHash, _signature);
 
         // Ensure the provided principal signed the expected message hash
-        require(recoveredSigner == _principal, "Unauthorized.");
+        if (recoveredSigner != _principal) revert Unauthorized();
 
         // Calculate salt
         uint256 salt = uint256(uint160(_principal)) + currectNonce;
```
Comment (rahul0eth) [inline src/PredictiveDeployer.sol]: Fix variable name from "currectNonce" to "currentNonce".

File: LICENSE [hunk 3]
```diff
@@ -1,6 +1,6 @@
 MIT License
```
Comment (rahul0eth) [inline LICENSE:1]: Add ".md" file extension to filename.

**Review by rahul0eth — CHANGES_REQUESTED** [review]
See comments.

**Review by cucupac — APPROVED** [review]

**Review by rahul0eth — APPROVED** [review]

**Pull request merged**